lib/inputstreamhelper/__init__.py

```diff
@@ -540,6 +549,37 @@ def _chromeos_config(self):
 
         return devices
 
+    def _remove_old_backups(self, backup_path):
+        """Removes old Widevine backups, if number of allowed backups is exceeded"""
+        backups = int(get_setting('backups', '4'))
+        versions = sorted([LooseVersion(version) for version in os.listdir(backup_path)])
+
+        if 'x86' in self._arch():
+            installed_version = self._load_widevine_config()['version']
+        else:
+            installed_version = self._select_best_chromeos_image(self._load_widevine_config())['version']
+
+        while len(versions) > backups + 1:
+            remove_version = str(versions[0])
+            if remove_version == installed_version:
+                try:
+                    remove_version = str(versions[1])
+                except IndexError:
+                    break
+
+            log('removing oldest backup which is not installed: {version}', version=remove_version)
+            shutil.rmtree(os.path.join(backup_path, remove_version))
+            versions = sorted([LooseVersion(version) for version in os.listdir(backup_path)])
+
+    def _install_cdm_from_backup(self, version):
+        """Copies files from specified backup version to cdm dir"""
+        filenames = os.listdir(os.path.join(self._backup_path(), version))
+
+        for filename in filenames:
+            xbmcvfs.copy(os.path.join(self._backup_path(), version, filename), os.path.join(self._ia_cdm_path(), filename))
+        log('Installed CDM version {version} from backup', version=version)
+        self._remove_old_backups(self._backup_path())
```

Comment:
I know that it makes it a little bit more complex, but I would remove first before adding the new. This will also lower the risk of a disk full condition.

Reply:
If it did that with allowed backups set to 0 and something went wrong, we'd have a situation like in #218, especially when using links. I thought I was supposed to fix that?

Reply:
No, because we wouldn't be removing the library we are using. But you are right that it wouldn't work as people expect when backups is set to 0.

Personally I would always leave one backup (the current library) behind so keep the lower limit to 1.